tests/tasks/test_datachecks.py

```diff
@@ -0,0 +1,165 @@
+            self.run_task,
+            '/haiti/tasks/check_person_data_validity', method='POST')
+
+
+class NoteDataValidityCheckTaskTests(task_tests_base.TaskTestsBase):
+    """Tests the note validity datacheck task.
+
+    Doesn't test the missing entry date or missing Person record ID case,
+    because those fields are required by the model class.
+    """
+
+    def init_testbed_stubs(self):
+        self.testbed.init_user_stub()
+        self.testbed.init_datastore_v3_stub()
+        self.testbed.init_taskqueue_stub()
+
+    def setUp(self):
+        super(NoteDataValidityCheckTaskTests, self).setUp()
+        self.data_generator.repo(repo_id='haiti')
+        self.person = self.data_generator.person()
+
+    def test_good_data(self):
+        self.data_generator.note(person_id=self.person.record_id)
+        self.run_task('/haiti/tasks/check_note_data_validity', method='POST')
+
+    def test_missing_original_creation_date(self):
+        self.data_generator.note(
+            person_id=self.person.record_id, original_creation_date=None)
+        self.assertRaises(
+            tasksmodule.datachecks.DatacheckException,
+            self.run_task,
+            '/haiti/tasks/check_note_data_validity', method='POST')
+
+    def test_invalid_author_email(self):
+        self.data_generator.note(
+            person_id=self.person.record_id,
+            author_email='not-a-valid-email-address')
+        self.assertRaises(
+            tasksmodule.datachecks.DatacheckException,
+            self.run_task,
+            '/haiti/tasks/check_note_data_validity', method='POST')
+
+    def test_invalid_email_of_found_person(self):
+        self.data_generator.note(
+            person_id=self.person.record_id,
+            email_of_found_person='not-a-valid-email-address')
+        self.assertRaises(
+            tasksmodule.datachecks.DatacheckException,
+            self.run_task,
+            '/haiti/tasks/check_note_data_validity', method='POST')
+
+    def test_missing_person_record(self):
+        self.data_generator.note(person_id='not-an-existing-person-record')
+        self.assertRaises(
+            tasksmodule.datachecks.DatacheckException,
+            self.run_task,
+            '/haiti/tasks/check_note_data_validity', method='POST')
+
+
+class ExpiredPersonRecordCheckTaskTest(task_tests_base.TaskTestsBase):
+
+    _NOW = datetime.datetime(2010, 2, 1)
+
+    def init_testbed_stubs(self):
+        self.testbed.init_user_stub()
+        self.testbed.init_datastore_v3_stub()
+        self.testbed.init_taskqueue_stub()
+
+    def setUp(self):
+        super(ExpiredPersonRecordCheckTaskTest, self).setUp()
+        utils.set_utcnow_for_test(ExpiredPersonRecordCheckTaskTest._NOW)
+        self.data_generator.repo(repo_id='haiti')
+
+    def test_good_expired_record(self):
+        expiry_date = (
+            ExpiredPersonRecordCheckTaskTest._NOW -
+            datetime.timedelta(days=1, hours=1))
```

Comment:
since you use it a few times, maybe make it another constant like _YESTERDAY or _PAST?

Reply:
Done.